tests/rancher-istio/rancher_istio.go

```diff
@@ -0,0 +1,947 @@
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"cni.repository", "testRepo/mirrored-istio-install-cni",
+				).
+				SetValue(
+					"cni.tag", "1.7.3",
+				),
+		},
+
+		{
+			Name: "Set .Values.cni.logLevel and .Values.cni.excludeNamespaces",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"cni.logLevel", "debug",
+				).
+				SetValue(
+					"cni.excludeNamespaces", "testNamespace",
+				).
+				SetValue(
+					"cni.excludeNamespaces", "testNamespace2",
+				),
+		},
+
+		{
+			Name: "Set .Values.pilot.enabled to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"pilot.enabled", "true",
+				),
+		},
+
+		{
+			Name: "Set .Values.pilot.enabled to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"pilot.enabled", "false",
+				),
+		},
+
+		{
+			Name: "Set .Values.pilot.repository and .Values.pilot.tag",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"pilot.repository", "testRepo/mirrored-istio-pilot",
+				).
+				SetValue(
+					"pilot.tag", "1.7.3",
+				),
+		},
+
+		{
+			Name: "Set .Values.pilot.hpaSpec",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"pilot.hpaSpec", testHPASpec,
+				),
+		},
+		{
+			Name: "Set .Values.pilot.podDisruptionBudget",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"pilot.podDisruptionBudget", testPodDisruptionBudget,
+				),
+		},
+
+		{
+			Name: "Set .Values.egressGateways.enabled to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"egressGateways.enabled", "true",
+				),
+		},
+
+		{
+			Name: "Set .Values.egressGateways.enabled to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"egressGateways.enabled", "false",
+				),
+		},
+
+		{
+			Name: "Set .Values.ingressGateways.type",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"egressGateways.type", "testType",
+				),
+		},
+
+		{
+			Name: "Set .Values.egressGateways.hpaspec",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"egressGateways.hpaspec", testHPASpec,
+				),
+		},
+
+		{
+			Name: "Set .Values.egressGateways.podDisruptionBudget",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"egressGateways.podDisruptionBudget", testPodDisruptionBudget,
+				),
+		},
+
+		{
+			Name: "Set .Values.ingressGateways",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"ingressGateways.enabled", "true",
```

Comment:
Why is this one test case but the `egressGateway` test case split into many?

Reply:
yeah merged all egress cases into one.